### PR TITLE
feat: add network status react hook

### PR DESCRIPTION
**Pull request by kingsleyzissou**
### Description
- Added a network status change hook to detect network changes.

##### Checklist

- [x] `npm test` passes
- [x] `npm run build` works
- [ ] tests are included
- [ ] documentation is changed or added [TODO]

### CODE REVIEW

**Comment by wtrocki** [comment]
Good bless you!

**Comment by wtrocki** [comment]
@kingsleyzissou Love this change! We do not have this documented for non hook version as well. 

Quick question of the purpose of this change. Generally, we will be focusing on bringing any network-related decision to offix internals. For example #49

I would like to know how you see this being used in sample app etc.

**Comment by kingsleyzissou** [comment]
Yeah I will add usage notes to the docs, was rushing the PR so I could get to a meeting. There is a small usage not on how to use it in the doc comment for the function.

Essentially, it came out of the comment where you said it would be better to add the network status listener to the config, but didn't really work for the example app in React.

**Comment by kingsleyzissou** [comment]
@darahayes I managed to get the tests working. I had to change the extension for the test to .tsx and everything then compiled.

File: packages/react-offix-hooks/test/useNetworkStatus.test.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  test("useNetworkStatus hook should return true initially", async () => {
+    const { result, waitForNextUpdate } = renderHook(() => useNetworkStatus(), { wrapper });
+    await waitForNextUpdate();
+    expect(result.current).toBe(true);
+  });
+  
+  test("useNetworkStatus hook status should change from true to false when network status changes", async () => {
+    const { result, waitForNextUpdate } = renderHook(() => useNetworkStatus(), { wrapper });
+    await waitForNextUpdate();
+    expect(result.current).toBe(true);
```
[the remark this answers is not in the document]
Comment (darahayes) [reply packages/react-offix-hooks/test/useNetworkStatus.test.tsx]: In this specific case you could take most of the code inside the `beforeAll` function and just place it inside a regular function or maybe two functions (one to build the client and one to build the wrapper) like this:

```ts
test('useNetworkStatus hook should return false initially', async () => {
  const client = await createClient({ online: false })
  const wrapper = createWrapper(client)

  // then do your regular test code
})
```

where 

* `createClient` is a simple function declared at the top of the file that sets up the client with the desired initial network status
* createWrapper is also a simple function declared at the top of the file that returns the wrapper function.

File: docs/ref-react-offix-hooks.md [hunk 2]
```diff
@@ -240,4 +240,20 @@ After the offline mutation cannot be replayed
     "networkError": {},
     "message": "Network error: Failed to fetch"
   }
+```
+
+## useNetworkStatus
+
+In order to detect changes in the network status, the useNetworkStatus hook can be used. The hook returns truth for when the network is online and falsey for when the network is offline. The hook makes use of a network status change listener and will update the returned value on a network status change event.
```
[the remark this answers is not in the document]
Comment (kingsleyzissou) [reply docs/ref-react-offix-hooks.md]: @darahayes yeah that makes a lot of sense. +100 for simpler language over academic style. The suggestion you've included is spot on

File: packages/react-offix-hooks/test/useNetworkStatus.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kingsleyzissou) [reply packages/react-offix-hooks/test/useNetworkStatus.test.tsx]: @darahayes thanks for that. Really solid feedback. I've made the changes to that and included those helper functions now to build the client and the wrapper.

**Pull request merged**